ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/operations/Deposit.java

```diff
@@ -27,16 +22,29 @@
 import tech.pegasys.artemis.util.hashtree.HashTreeUtil.SSZTypes;
 import tech.pegasys.artemis.util.hashtree.Merkleizable;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
 public class Deposit implements Merkleizable {
 
   private List<Bytes32> proof; // Bounded by DEPOSIT_CONTRACT_TREE_DEPTH
   private DepositData data;
+  private UnsignedLong index;
```

Comment:
This is not in spec, so I think its a bit confusing to add client-specific information to the data structure. Maybe we should do this similar to BeaconStateWithCache, where you have a child class that inherits Deposit and has fields to keep client-specific or non-spec information.

Reply:
This is necessary for creating the Deposit Merkle tree. Without it we have no way of ordering the deposits in the tree.

Reply:
I understand its necessary. I’m just saying that you can achieve the same thing by having an object that inherits the deposit object and keeps the necessary extra information, instead of modifying the Deposit object itself.

Reply:
https://github.com/PegaSysEng/artemis/issues/742